packages/mobile/src/account/SupportContact.tsx

```diff
@@ -83,13 +106,26 @@ const SupportContact = (props: Props) => {
       }
     }
     setInProgress(false)
-    Mailer.mail(email, (error: any, event: string) => {
+
+    // Try to send with native mail app with logs as attachment
+    // if fails user can choose mail app but logs sent in message
+    Mailer.mail(email, async (error: any, event: string) => {
       if (event === 'sent') {
-        navigate(Screens.Account)
+        navigateBack()
         props.showMessage(t('contactSuccess'))
-      }
-      if (error) {
-        Logger.showError(error + ' ' + event)
+      } else if (error) {
+        const emailSent = await sendEmailWithNonNativeApp(
+          emailSubject,
+          message,
+          deviceInfo,
+          combinedLogsPath
+        )
+        if (emailSent.success) {
+          navigateBack()
+          props.showMessage(t('contactSuccess'))
```

Comment:
Can we have it as a separate method to avoid code duplication?

Reply:
Done!